### PR TITLE
barannikova_m_LABML1

### PR DESCRIPTION
**Pull request by MashkaBM**

### CODE REVIEW

File: barannikova_m_LABML1.ipynb [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Wall time: 27.1 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "clf = LogisticRegression(random_state=42)\n",
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:329]: Мне кажется, имело смысл время обучения для каждой модели отдельно посмотреть. Ну, вот эта вот `%%time`, сейчас в этой клетке все три модели обучаются.

File: barannikova_m_LABML1.ipynb [hunk 2]
```diff
@@ -0,0 +1,1135 @@
...
+    "plt.xlabel('n_neighbours')\n",
+    "plt.title('Accuracy')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "оптимальным явл n=3"
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:429]: Почему не 1?) А какое качество для 1 соседа? Не 100%? (должно ведь быть 100 по идее?) Можно было ещё один "слой усреднения" добавить, проведя несколько раз разбиение на train-test (а то, может, для двух соседей просто разбиение "неудачным" оказалось).

File: barannikova_m_LABML1.ipynb [hunk 3]
```diff
@@ -0,0 +1,1135 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 197,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "\n",
+    "max_depth = np.arange(15, 25).astype('int')\n",
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:458]: Почему от глубины 15 начинаете перебирать? Может, имело смысл маленькие захватить ещё? (чтоб совсем деревья "простенькие" были)

File: barannikova_m_LABML1.ipynb [hunk 4]
```diff
@@ -0,0 +1,1135 @@
...
+    "from sklearn.ensemble import GradientBoostingClassifier"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 203,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model = GradientBoostingClassifier(random_state=42, n_estimators=3, max_depth=23, subsample=0.3).fit(X_train, y_train)\n",
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:525]: С бустингом никаких гиперпараметров не стали подбирать?)

File: barannikova_m_LABML1.ipynb [hunk 5]
```diff
@@ -0,0 +1,1135 @@
...
+    "**Пункт 2.1:**Как вы думаете, можно ли для данной задачи в качестве метрики качества использовать Accuracy? Обоснуйте свой ответ"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "В качестве метрики accuracy будет корректно использовать только для логистической регрессии, для случаев с линейной регрессией ridge и lasso данную метрику не стоит использовать, поскольку она показывает долю правильно предсказанных ответов на задачу классификации. А в случае линейной регресии мы даже близкие к правильным цифрам будем отвергать, поскольку часто получаем нецелые числа (округление не поможет, тк неясно, в какую сторону округлить)."
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:542]: Может, просто именно Ridge и Lasso тогда не совсем корректно применять для классификации?) Мне кажется, в вопросе имели в виду, корректно ли Accuracy в принципе применять в этой задаче, где надо научиться классифицировать по цифрам MNIST... Ну, то есть, точно ли для логистической регрессии корректно Accuracy использовать?

File: barannikova_m_LABML1.ipynb [hunk 6]
```diff
@@ -0,0 +1,1135 @@
...
+   "metadata": {},
+   "source": [
+    "**Пункт 2.2:** в любом случае, посмотрите на Accuracy для ваших моделей. Какая из них отработала лучше всего? Как вы думаете, почему?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Как и ожидалось, метрика accuracy корректно работает только дл логистической регрессии, причем доля правильных ответов составляет больше 90%, что говорит о неплохом качестве модели"
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:556]: А для логистической регрессии вроде не считали вообще accuracy? Только для KNN, леса и бустинга были графики?

File: barannikova_m_LABML1.ipynb [hunk 7]
```diff
@@ -0,0 +1,1135 @@
...
+    "    confusion_matrix\n",
+    ")\n",
+    "\n",
+    "accuracy_logic = accuracy_score(y_test, predicted_logic)\n",
+    "\n",
+    "#accuracy_ridge = accuracy_score(y_test, predicted_ridge)\n",
+    "#accuracy_lasso = accuracy_score(y_test, predicted_lasso)\n",
+    "\n",
+    "\n",
+    "print(accuracy_logic.round(3))"
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:584]: А, всё, да, теперь посчитали)

File: barannikova_m_LABML1.ipynb [hunk 8]
```diff
@@ -0,0 +1,1135 @@
...
...Nz9CGtgIbgxhak6M1cuIX0CP9tPErdrn97Wtwj9EEn5dLwX7sChfXObwEkGtRRDpXiljmp+vqp6T2KXcdzM6DeZ9JZ9aojD4KjKJAhEjkXoOSj9NnEUoZg5chdeJ3BGhNYASUvh9ZoMNu8SBx7p+UvIqaQY6g26B1FGIKcyxFFEamYJUVIYrH40Pqazn9gbpOrzBwar0weeXLMYlSmMyhSpqUWiwEVJZc8UAoIokp49R6q+QP/eh8RhgG/3UM0sRmkaBIHUMQsNAKIQvTrDcO0WAKKqE4UBURBgN3fQS1MIgoA37OE2d4jCAK1QGXuQHUeUZMzpxVPbH0TWU6CnsHbX6a/eGDf0a6UpcuevIR426xeruK1dAmeULPtKCsb0Iv6gQ+TYiLKKUZkntC1ESSZ78Q3iwEtC1D0nsSfpNgh9F61QJfK9xPC3MoNROX3B7d55l/7dI48ue3cDa2+D4itfODZ0cB+/30JJ54mB4cYdhCd83Q8iCAKXpy4yV5jF8R2yRua5rTI+TbiOzaDbRBJlMoXyeKkw8H1Wb7xLay8JBkcQmFm6hKobdBu7dErVT0wQFSpTFCovb/n6RRE4DqKqvhQftRfFxt2PsPaG9NdaRGFE5//P3n/FSpavWX7Yb3sX3h3v0md5c/3tnh5KpNikOKCkB0kUBxAgM6IoQQT0IAdCAiQB0ptEyFAaDfUiQeCThhwNOBzOYNAz3X2nb9+ue8tXupN5vIkTPra3ethx4pzIczIrs6qyKqtuLuDi1tkZZu8dEXuv//etb637bZJhzNrP3vi90ljJhkpoz+pvZVV56jlIwpjxfh+/5yBrCtZS+YW2gUc7XYJRQOyHSJpMJsB4t0f91lfPfn0Z8bxz1K9q5S8RFKuMbFhEvot3sJVHEE0g6RaSquOf7OF1DiED92gHuVij9sbPCDoHpGGA3zvCWFiHcHa1lOdWdrks5ivoHtN8/6+jNRZInDGSWUA7J/jPq09P1x/Fnsvg3m+JnTGSolG58hZKsYJiFdFq8zN2GWqxmod6CyLFjdtE9hD3ZA9VqpK4Y4b3PiRaWEOvz9P74jcwCQ33jncpbbyGubA2895pFBI5IyRNRzYKeJ1DwmEHQZTRm4uohbO2WhL6jLfvTAlPGof0Pvs1bnsXrVjDXFjFnFsljSP87jHB4AS91kJrLuLsbSJNwshly8onL2UlH
```
Comment (Alvant) [inline barannikova_m_LABML1.ipynb:780]: 🤩

**Review by Alvant — APPROVED** [review]
